exec/cnex/gc.c

```diff
@@ -0,0 +1,73 @@
+#include "gc.h"
+
+#include <assert.h>
+#include <inttypes.h>
+#include <stdlib.h>
+
+#include "bytecode.h"
+#include "cell.h"
+#include "exec.h"
+#include "framestack.h"
+#include "module.h"
+#include "nstring.h"
+#include "stack.h"
+#include "util.h"
+
+
+
+Cell *heap_allocObject(TExecutor *exec)
+{
+    TAllocationObject *o = malloc(sizeof(TAllocationObject));
+    if (o == NULL) {
+        fatal_error("Failed to allocate heap object after %ul allocations.", exec->allocations);
+    }
+    exec->diagnostics.total_allocations++;
+
+    Cell *p = cell_newCell();
+    o->allocNum = exec->diagnostics.total_allocations;
+    o->object = p;
+    o->next = NULL;
+
+    // Push newaly allocated object on the virtual heap.
+    if (exec->firstObject != NULL) {
```

Comment:
You don't have to check to see whether `exec->firstObject` is null or not before assigning it. Also, since you don't need to check here, you don't need the assignment on line 29 either.

Reply:
Fixed logic around this.